From 2af1e67be09516718d9174eacd3d3e54d705fe46 Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <shengjiu.wang@nxp.com>
Date: Tue, 23 Jan 2018 13:25:40 +0800
Subject: [PATCH 3237/5242] MLK-17442: ASoC: fsl: fix wrong usage of
 filter_data

commit  5204d55a5bba452fef777f519ec7973f7fdec51f from
https://source.codeaurora.org/external/imx/linux-imx.git

The filter_data should be used for dma_filter_fn function,
but we used the filter_data wrongly for dma channel name.
This patch is to fix the issue.

Signed-off-by: Shengjiu Wang <shengjiu.wang@nxp.com>
Reviwed-by: Daniel Baluta <daniel.baluta@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sound/soc/fsl/fsl_esai.c       |    4 ++--
 sound/soc/fsl/fsl_sai.c        |    4 ++--
 sound/soc/fsl/imx-pcm-dma-v2.c |    2 +-
 3 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/sound/soc/fsl/fsl_esai.c b/sound/soc/fsl/fsl_esai.c
index 6b1588b..fb73816 100644
--- a/sound/soc/fsl/fsl_esai.c
+++ b/sound/soc/fsl/fsl_esai.c
@@ -1061,8 +1061,8 @@ static int fsl_esai_probe(struct platform_device *pdev)
 	else
 		esai_priv->fifo_depth = 64;
 
-	esai_priv->dma_params_rx.filter_data = "rx";
-	esai_priv->dma_params_tx.filter_data = "tx";
+	esai_priv->dma_params_rx.chan_name = "rx";
+	esai_priv->dma_params_tx.chan_name = "tx";
 	esai_priv->dma_params_tx.maxburst = 16;
 	esai_priv->dma_params_rx.maxburst = 16;
 	esai_priv->dma_params_tx.addr = res->start + REG_ESAI_ETDR;
diff --git a/sound/soc/fsl/fsl_sai.c b/sound/soc/fsl/fsl_sai.c
index f45a33d..1676840 100644
--- a/sound/soc/fsl/fsl_sai.c
+++ b/sound/soc/fsl/fsl_sai.c
@@ -1123,8 +1123,8 @@ static int fsl_sai_probe(struct platform_device *pdev)
 				   MCLK_DIR(index));
 	}
 
-	sai->dma_params_rx.filter_data = "rx";
-	sai->dma_params_tx.filter_data = "tx";
+	sai->dma_params_rx.chan_name = "rx";
+	sai->dma_params_tx.chan_name = "tx";
 	sai->dma_params_rx.addr = res->start + FSL_SAI_RDR0;
 	sai->dma_params_tx.addr = res->start + FSL_SAI_TDR0;
 	sai->dma_params_rx.maxburst = FSL_SAI_MAXBURST_RX;
diff --git a/sound/soc/fsl/imx-pcm-dma-v2.c b/sound/soc/fsl/imx-pcm-dma-v2.c
index 6dfeeda..2114c4a5 100644
--- a/sound/soc/fsl/imx-pcm-dma-v2.c
+++ b/sound/soc/fsl/imx-pcm-dma-v2.c
@@ -115,7 +115,7 @@ static int imx_pcm_open(struct snd_pcm_substream *substream)
 		struct dma_chan *chan;
 
 		chan = dma_request_slave_channel(rtd->cpu_dai->dev,
-						 dma_data->filter_data);
+						 dma_data->chan_name);
 		ret = snd_dmaengine_pcm_open(substream, chan);
 		if (ret)
 			return ret;
-- 
1.7.9.5

